clang/examples/DPCT/Thrust/thrust$$set_union_by_key.cu

```diff
@@ -0,0 +1,140 @@
+#include <thrust/device_malloc.h>
```

Comment:
Prelease add lit test to see what is the migration result of set_union_by_key

Reply:
Here is the lit test: https://github.com/oneapi-src/SYCLomatic/blob/73874101f128eaecd0a14f5f52ddf9a1b2723a9f/clang/test/dpct/thrust-algo-part3.cu#L245

Reply:
ref to https://github.com/oneapi-src/SYCLomatic/pull/1195/files#diff-a1b8a2634fb9afbfb3d5eadc1a79f4dc6af2aa8717e72c20dc42807479428ad5